scripts/pushit.sh

```diff
@@ -1,65 +1,101 @@
 #!/usr/bin/env bash
-
+# Copy outputs of a process into a subdirectory representing a git branch and
+# push that branch up
+#
+# USAGE
+#
+#   bash pushit.sh <branch> <repo> <slug> <email> <token>
+#
+# ARGUMENTS
+#
+#   branch  the branch the output directory represents (defaults to `gh-pages`)
+#   repo    the repository to push to
+#   slug    slug of the pusher (e.g. hubdashboard[bot] or zkamvar)
+#   email   email of the pusher (e.g. 186236658+hubdashboard[bot]@users.noreply.github.com)
+#   token   GitHub Token used for authentication, must have `contents: write scope`
+#             associated with the <email>
+#
+# SETUP
+#
+#   Before this script is run, you must have the following in place:
+#
+#   1. a local clone of <repo>@<branch> in a directory called either "pages"
+#      (for branch=gh-pages) or "data" for others
+#   2. output of a previous build process _in the current working directory_.
+#
 branch="${1:-'gh-pages'}"
 repo="${2:-missing}"
 slug="${3:-missing}"
 email="${4:-missing}"
 token="${5:-missing}"
 
-
+# There are currently three branches that this script knows what to do with.
+#
+# There are four variables that we set based on the branches:
+#
+#  - dir      the directory of the cloned branch
+#  - msg      the commit message
+#  - amend    a boolean that indicates if the commit should be amended (for
```

Comment:
I don't understand why we do amend commits for gh-pages?

Reply:
I believe I was using this as a way of reducing the size of the branch